### PR TITLE
update input to `4.2.0` #38

### PR DESCRIPTION
**Pull request by sun-mota**
# Alaska Airlines Pull Request

- This PR is to update input to auro-input@4.2.0 which has a new registration method
- It is part of #21 but will merge into `sunMota/componentIntegration` branch 
- This has new rollup.config to create bundled min.js file for `demo` scripts. But this script does not affect the functionality of the components

**By submitting this Pull Request, I confirm that my contribution is made under the terms of the Apache 2.0 license and I have performed a self-review of my own update.**

## Summary by Sourcery

Update auro-input to version 4.1.3, introducing a new registration method and updating dependencies. Enhance build scripts and documentation to support these changes.

New Features:
- Introduce a new registration method for the auro-input component, allowing it to be registered with a custom name.

Enhancements:
- Update auro-input to version 4.1.3, which includes improvements in the registration method and dependency updates.

Build:
- Update build scripts to include new tasks for building, testing, and documentation generation.

Documentation:
- Update documentation to reflect the new registration method for auro-input and correct minor typos.

### CODE REVIEW

File: .stylelintrc [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
```
Comment (jordanjones243) [inline .stylelintrc:1]: This file should not be edited and should stay consistent with the generator/all other components:
- https://github.com/AlaskaAirlines/WC-Generator/blob/master/template/.stylelintrc

Comment (jordanjones243) [reply .stylelintrc:1]: If a style rule is failing, either update the code or add a `/* stylelint-disable (rule) */ in the file where the issue is.

See for example:
https://github.com/AlaskaAirlines/auro-button/blob/e431a34a4e93b47ddd22c2e014c264580c28a116/src/style.scss#L6

File: components/dropdown/demo/api.min.js [hunk 2]
```diff
@@ -9,28 +10,27 @@ function showExample() {
...
   const dropdown = document.querySelector('#hideExample');
+
   btn.addEventListener('click', () => {
     dropdown.hide();
   });
 }
 
 /* eslint-disable jsdoc/require-jsdoc, no-magic-numbers */
-function initExamples(initialCount = 0) {
-  const MAX_RETRY_ATTEMPTS = 20;
```
Comment (jordanjones243) [inline components/dropdown/demo/api.min.js:19]: Is there a reason you removed these global variables?

File: components/input/package.json [hunk 3]
```diff
@@ -148,17 +147,44 @@
...
     "demo:update:index": "node ./scripts/prepForBuild",
     "dev": "concurrently --kill-others 'npm run build:watch' 'npm run serve'",
     "dist:js": "copyfiles -u 1 -V './src/**/*.js' ./dist",
+    "esLint": "./node_modules/.bin/eslint src/**/*.js --ignore-pattern 'src/**/*Version.js'",
+    "linters": "npm-run-all scssLint esLint",
+    "preCommit": "node ./node_modules/@aurodesignsystem/auro-library/scripts/build/pre-commit.mjs",
     "postCss:component": "node ./node_modules/@aurodesignsystem/auro-library/scripts/build/postCss.mjs",
-    "postinstall": "node ./node_modules/@aurodesignsystem/auro-library/scripts/build/postinstall.mjs",
-    "sass:render": "sass-render --load-path=node_modules 'src/**/*.css' 'components/**/*.css' -t ./node_modules/@aurodesignsystem/auro-library/scripts/build/staticStyles-template.js",
+    "postinstall": "node packageScripts/postinstall.mjs",
```
Comment (jordanjones243) [inline components/input/package.json:176]: We should be using the auro-library scripts

File: components/input/package.json [hunk 4]
```diff
@@ -148,17 +147,44 @@
...
+    "sass:render": "sass-render 'src/**/*.css' 'components/**/*.css' -t ./node_modules/@aurodesignsystem/auro-library/scripts/build/staticStyles-template.js",
     "serve": "web-dev-server --open demo/ --node-resolve --watch",
-    "types": "tsc"
+    "sweep": "rm -rf ./dist | find ./demo ./dist ./src -type f \\( -name \"*.css\" -o -name \"*-css.js\" \\) -delete",
+    "test": "wtr --coverage",
+    "test:watch": "wtr --watch",
+    "prepare": "husky install",
+    "vrt:approve": "backstop approve",
```
Comment (jordanjones243) [inline components/input/package.json:183]: Could you explain what these `vrt` scripts are doing?

File: components/input/test/auro-input.test.js [hunk 5]
```diff
@@ -67,19 +67,8 @@ describe('auro-input', () => {
     `);
 
     const clearButton = el.shadowRoot.querySelector('.clearBtn');
-    const input = el.shadowRoot.querySelector('input');
```
Comment (jordanjones243) [inline components/input/test/auro-input.test.js:70]: Why were these parts removed from the test?

File: eslint.config.mjs [hunk 6]
```diff
@@ -19,7 +19,8 @@ export default [
     'components/**/*Version.js',
     'components/**/apiExamples/**',
     'components/**/test/**',
-    'components/**/demo/**'
+    'components/**/demo/**',
+    'components/**/dist/**'
```
Comment (jordanjones243) [inline eslint.config.mjs:23]: Is there a reason you added the `dist` directory?

File: package.json [hunk 7]
```diff
@@ -20,9 +20,9 @@
     "node": "^18.x || ^20.x "
   },
   "dependencies": {
-    "@aurodesignsystem/auro-button": "^8.1.2",
+    "@aurodesignsystem/auro-button": "^8.1.3",
     "@aurodesignsystem/auro-formvalidation": "^1.0.3",
-    "@aurodesignsystem/auro-icon": "^5.0.2",
+    "@aurodesignsystem/auro-icon": "^6.0.2",
```
Comment (jordanjones243) [inline package.json:25]: Icon has new version release.

**Review by jordanjones243 — CHANGES_REQUESTED** [review]

File: .stylelintrc (hunk 1 above)
Comment (sun-mota) [reply .stylelintrc:1]: this file is copied from https://github.com/AlaskaAirlines/auro-input/blob/main/.stylelintrc let me know if there is any other file that we want to copy from.

File: components/dropdown/demo/api.min.js (hunk 2 above)
Comment (sun-mota) [reply components/dropdown/demo/api.min.js:19]: this is generated from the bundler.

File: components/input/package.json (hunk 3 above)
Comment (sun-mota) [reply components/input/package.json:176]: Currently no script that I use as a reference and also it's unclear what script will be remained after Turbo. I'd like to keep it as Auro-input until we know what we can do with Turbo.

File: components/input/package.json (hunk 4 above)
Comment (sun-mota) [reply components/input/package.json:183]: it's copied from the current auro-input. I have no idea about the legacy code. Should we remove?

File: components/input/test/auro-input.test.js (hunk 5 above)
Comment (sun-mota) [reply components/input/test/auro-input.test.js:70]: I'm guessing because it was not being used?

File: eslint.config.mjs (hunk 6 above)
Comment (sun-mota) [reply eslint.config.mjs:23]: eslint is working on bundler generated file. any reason why we want to lint the bundled file?

File: package.json (hunk 7 above)
Comment (sun-mota) [reply package.json]: updated

File: components/input/test/auro-input.test.js (hunk 5 above)
Comment (sun-mota) [reply components/input/test/auro-input.test.js:70]: this file is just copied from the latest `auro-input`

File: components/input/package.json (hunk 4 above)
Comment (jordanjones243) [reply components/input/package.json:183]: You can keep this.

File: .stylelintrc (hunk 1 above)
Comment (sun-mota) [reply .stylelintrc]: done! :)

File: components/input/package.json (hunk 3 above)
Comment (sun-mota) [reply components/input/package.json]: reverted the scripts section

File: eslint.config.mjs (hunk 6 above)
Comment (sun-mota) [reply eslint.config.mjs:23]: image

Comment (jordanjones243) [reply eslint.config.mjs:23]: You should remove the `dist` folder from the linters.

Comment (jordanjones243) [reply eslint.config.mjs:23]: Yes we can ignore these files and will mark as resolved.

File: components/input/src/auro-input.js [hunk 8]
```diff
@@ -13,10 +13,12 @@ import i18n from './i18n.js';
 import BaseInput from './base-input.js';
 
 import { AuroDependencyVersioning } from '@aurodesignsystem/auro-library/scripts/runtime/dependencyTagVersioning.mjs';
+import AuroLibraryRuntimeUtils from '@aurodesignsystem/auro-library/scripts/utils/runtimeUtils.mjs';
+
 import { AuroIcon } from '@aurodesignsystem/auro-icon/src/auro-icon.js';
-import iconVersion from './iconVersion.js';
+import iconVersion from '../../../src/iconVersion.js';
```
Comment (jordanjones243) [inline components/input/src/auro-input.js:19]: There is also a `components/input/src/buttonVersion.js` and `iconVersion.js`. Do we still need these files if we are pointing to the root level files?

**Review by jordanjones243 — CHANGES_REQUESTED** [review]

**Review by jordanjones243 — CHANGES_REQUESTED** [review]
Run `npm run build:release` from root level, there are some file changes

**Comment by jordanjones243** [comment]
Looks like input got a new release and these changes need to be added to the PR:
https://github.com/AlaskaAirlines/auro-input/commit/3abd010f2598db5428c2cab6c4716f8ac7e46a1a#diff-247b6ba6752b3b29ab8b50abe8fe8c0c157ad49f85bce9b52a605cc2edb4fa69

File: components/input/src/auro-input.js (hunk 8 above)
Comment (sun-mota) [reply components/input/src/auro-input.js:19]: removed `components/input/src/*Version.js`

File: components/input/docs/partials/api.md [hunk 9]
```diff
@@ -557,6 +557,51 @@ Use the `type="year-month-day"` attribute for a date formatted input.
 
 </auro-accordion>
 
+Use the `type="month"` attribute for a date formatted input.
```
Comment (jordanjones243) [inline components/input/docs/partials/api.md:560]: Looks like the examples got added but they don't appear on the page. When running a build, the `demo/api.md` file should have been updated as well. Looks like there is an issue with the building of the docs?

**Review by jordanjones243 — CHANGES_REQUESTED** [review]

**Review by jordanjones243 — APPROVED** [review]

File: components/input/docs/partials/api.md (hunk 9 above)
Comment (sun-mota) [reply components/input/docs/partials/api.md:560]: known issue. will work on https://github.com/AlaskaAirlines/auro-form/issues/30

**Pull request merged**